app/controllers/projects_controller.rb

```diff
@@ -89,4 +94,41 @@ def project_params
     end
     params['project']
   end
+  
+  private
+  def order_by_project_name
+    if session[:pre_click] == "project_name"
+      if session[:order] == "ASC"
+        @projects = Project.order(name: :desc)
+      else
+        @projects = Project.order(:name)
+      end
+    else    
+      @projects = Project.order(:name)  
+    end
+    change
+  end
+  
+  def order_by_metric_name 
+    click_type = params[:type]
+    if session[:pre_click] == click_type
+      if session[:order] == "ASC"
+        @projects = Project.joins(:metric_samples).where("metric_samples.metric_name = ?", click_type).order("metric_samples.score")
+      else
+        @projects = Project.joins(:metric_samples).where("metric_samples.metric_name = ?", click_type).order("metric_samples.score").reverse
+      end
+    else
+      @projects = Project.joins(:metric_samples).where("metric_samples.metric_name = ?", click_type).order("metric_samples.score")
+    end 
+    change
+  end
+  
+  def change
+    if session[:order] == "ASC"
```

Comment:
line 127-131 can be replaced with
`session[:order] = session[:order] == "ASC" ? "DESC" : "ASC"`